exercises/runLengthEncode/runLengthEncode.js

```diff
@@ -20,11 +20,24 @@
  * @returns {string} A run-length encoded copy of the input string
  */
 
-function runLengthEncode(num) {
-  // This is your job. :)
-
-  // Remember, if the code is stumping you, take a step back and
-  // make sure you can do it by hand.
+function runLengthEncode(string) {
+  let stringChars = [];
+  for (let char of string){
+    stringChars.push(char);
+  }
+  const encodeList = [];
+  while (stringChars.length){
+    let charCount = 1;
+    let char = stringChars.shift();
```

Comment:
This is a very expensive operation. You should run this through the benchmark code (see the `sessions` repository) and see what kind of curve you get as a function of the length of the input string. You'd be surprised.